aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1281,6 +1308,118 @@ std::tuple<Tensor, Tensor> _symeig_helper_cuda(const Tensor& self, bool eigenvec
   }
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ eig ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+template <typename scalar_t>
+static void apply_eig(const Tensor& self, bool eigenvectors, Tensor& out_eigvals, Tensor& out_eigvecs,
+                      int64_t *info_ptr) {
+#ifndef USE_MAGMA
+AT_ERROR("symeig: MAGMA library not found in "
```

Comment:
AT_ERROR is deprecated and this refers to symeig, not eig. 

```
TORCH_CHECK(false, "...");
```

I know the form of this error message is inherited, but it's terrible so let's update it. What about:

"Calling torch.eig on a CUDA tensor requires compiling PyTorch with MAGMA. Either transfer the tensor to the CPU before calling torch.eig or recompile with MAGMA."?

Reply:
fixed by 89ca976835. Note that all the other functions in the same file are
still using AT_ERROR and the old error message